flowclient/flowclient/client.py

```diff
@@ -1654,15 +1652,17 @@ def random_sample(
     Returns
     -------
     dict
-        Dict which specifies the parameters for the random sample.
+        Dict which functions as the query specification.
     """
-    spec = {
+    sampled_query = query.copy()
```

Comment:
Slightly more pythonic to do `dict(query)`, although obviously in both cases any nested dicts still refer to their underlying object.

Reply:
Ah, good idea. I'm hoping a shallow copy is sufficient since `random_sample` doesn't modify any existing values in `query`, just adds a new one.